src/clj/puppetlabs/services/file_serving/config/puppet_fileserver_config_core.clj

```diff
@@ -0,0 +1,93 @@
+(ns ^{:doc
+      "Tools to parse Puppet fileserver.conf."}
+    puppetlabs.services.file-serving.config.puppet-fileserver-config-core
+  (:require
+    [puppetlabs.kitchensink.core :as ks]
+    [schema.core :as schema]
+    [puppetlabs.puppetserver.ringutils :as ring])
+  (:use [clojure.java.io :only (reader)]
+        [clojure.string :only (split trim split)]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; internal helpers
+
+(defn- parse-one-line
+  [s]
+  (if (= (first s) \[)
+    (-> s (subs 1 (.indexOf s "]")) trim keyword)
```

Comment:
Style nitpick: I think in general, it's better for readability to format threading macros so that each form is on it's own line:

```clojure
    (-> s
        (subs 1 (.indexOf s "]"))
        trim
        keyword)
```